static/js/upload_widget.js

```diff
@@ -76,21 +76,22 @@ export function build_widget(
             const file = e.target.files[0];
             if (file.size > max_file_upload_size * 1024 * 1024) {
                 input_error.text(
-                    i18n.t("File size must be < __max_file_size__Mb.", {
-                        max_file_size: max_file_upload_size,
-                    }),
+                    $t(
+                        {defaultMessage: "File size must be < {max_file_size}Mb."},
+                        {max_file_size: max_file_upload_size},
+                    ),
```

Comment:
What else can go in this `defaultMessage` dictionary?  I feel like it'd be a lot cleaner in terms of how many of these get exploded to several lines if one could pass just the string for this first parameter.

(Or maybe they need that for extraction)?

Reply:
This is needed for [extraction](https://formatjs.io/docs/tooling/cli/#extraction), yeah. It’s a [message descriptor](https://formatjs.io/docs/intl/#message-descriptor) whose other fields are `id` and `description`. `id` is actually required, but it’s inserted automatically by the [babel-plugin-formatjs](https://formatjs.io/docs/tooling/babel-plugin/). `description` is an optional hint for the translator and can distinguish multiple messages that are the same in English but may translate differently.